sdk-base/src/main/java/com/mapbox/maps/extension/observable/ObservableExtension.kt

```diff
@@ -301,124 +299,329 @@ fun ObservableInterface.unsubscribeSourceRemoved(observer: Observer) =
+  val map = data.contents as Map<String, Value>
+  return SourceAddedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END),
+    id = map.nonNullString(ID)
+  )
 }
 
 /**
  * Get the parsed event data for source removed event.
  * @return a parsed SourceRemovedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getSourceRemovedEventData(): SourceRemovedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, SourceRemovedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return SourceRemovedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END),
+    id = map.nonNullString(ID)
+  )
 }
 
 /**
  * Get the parsed event data for render frame started event.
  * @return a parsed RenderFrameStartedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getRenderFrameStartedEventData(): RenderFrameStartedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, RenderFrameStartedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return RenderFrameStartedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END)
+  )
 }
 
 /**
  * Get the parsed event data for render frame finished event.
  * @return a parsed RenderFrameFinishedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getRenderFrameFinishedEventData(): RenderFrameFinishedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, RenderFrameFinishedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return RenderFrameFinishedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END),
+    renderMode = RenderMode.valueOf(map.nonNullString(RENDER_MODE).toUpperCase(Locale.US)),
+    needsRepaint = map.nonNullBoolean(NEEDS_REPAINT),
+    placementChanged = map.nonNullBoolean(PLACEMENT_CHANGED)
+  )
 }
 
 /**
  * Get the parsed event data for camera changed event.
  * @return a parsed CameraChangedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getCameraChangedEventData(): CameraChangedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, CameraChangedEventData::class.java)
-}
\ No newline at end of file
+  val map = data.contents as Map<String, Value>
+  return CameraChangedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END)
+  )
+}
+
+internal fun Map<String, Value>.nonNullLong(name: String): Long {
+  return this[name]!!.contents as Long
+}
+
+internal fun Map<String, Value>.nullableLong(name: String): Long? {
+  return this[name]?.contents as Long?
+}
+
+internal fun Map<String, Value>.nonNullString(name: String): String {
+  return this[name]!!.contents as String
+}
+
+internal fun Map<String, Value>.nullableString(name: String): String? {
+  return this[name]?.contents as String?
+}
+
+internal fun Map<String, Value>.nonNullInt(name: String): Int {
+  return (this[name]!!.contents as Long).toInt()
```

Comment:
```suggestion
  return (this[name]!!.contents as Int).toInt()
```

Reply:
`Value` only supports Long, so here we must use `Long` and then transform to `Int`